internal/common/environment.go

```diff
@@ -23,3 +28,16 @@ func EnvOrDefault(envName string, defVal string) string {
 func GetOperatorVersion() string {
 	return EnvOrDefault(OperatorVersionKey, defaultOperatorVersion)
 }
+
+func IsSingleNodeMode(topologyMode osconfv1.TopologyMode) bool {
+	return topologyMode == osconfv1.SingleReplicaTopologyMode
+}
+
+func GetInfrastructureTopology(c client.Client) (osconfv1.TopologyMode, error) {
```

Comment:
This function is only called from `ssp_operator.go`, can you move it there?
I know it is related to environment, but I think moving a function to a different package is useless, if it is not going to be used in multiple places.

Reply:
I was naively taught that files should be small and things should sit in seperate places depending on semantics. I see we do not do that much, and perhaps this is the Go way.
The logic behind moving stuff into seperate places early is that the chances someone else will move it later if it grows are smaller, since moving code is harder than just writing in the same place (for lazy people like me).
However I do not have strong feelings about this - if you think I should move it, I will, let me know.

Reply:
I don't know if it is a "Go way", it is probably just my code style, and I did not realize it.
This file does not have a clear purpose, I made it to contain environment variables, but it can have other code.

I think file size alone does not matter with current IDEs, I don't usually read the full file, I only jump to relevant functions. 
Separating based on semantics is a good idea. Also moving functions is easy with current IDEs. For example there is a functionality in GoLand that can move a function or structure to a different file and fix all dependencies in "one click".

You can keep this here.